src/ipc-handlers.ts

```diff
@@ -342,7 +342,17 @@ export async function startServer(
 	if ( parentWindow && ! parentWindow.isDestroyed() && ! event.sender.isDestroyed() ) {
 		parentWindow.webContents.send( 'theme-details-changed', id, server.details.themeDetails );
 	}
-	server.updateCachedThumbnail().then( () => sendThumbnailChangedEvent( event, id ) );
+
+	if ( server.details.running ) {
+		server
+			.updateCachedThumbnail()
+			.then( () => {
+				sendThumbnailChangedEvent( event, id );
+			} )
+			.catch( ( error ) => {
+				console.error( `Failed to update thumbnail for server ${ id }:`, error );
+			} );
```

Comment:
Maybe we could consider using try/catch for better readability:

```suggestion
		try {
			await server.updateCachedThumbnail();
			sendThumbnailChangedEvent( event, id );
		} catch ( error ) {
			console.error( `Failed to update thumbnail for server ${ id }:`, error );
		}
```